app/org/sagebionetworks/bridge/validators/SurveyValidator.java

```diff
@@ -132,11 +132,24 @@ private void validateRules(Errors errors, List<SurveyElement> elements) {
             SurveyElement element = elements.get(i);
             if (element instanceof SurveyQuestion) {
                 for (SurveyRule rule : ((SurveyQuestion)element).getConstraints().getRules()) {
-                    if (alreadySeenIdentifiers.contains(rule.getSkipToTarget())) {
-                        errors.pushNestedPath("elements["+i+"]");
+                    errors.pushNestedPath("elements["+i+"]");
+                    // Validate the rule either has a skipTo target, or an endSurvey = TRUE, but not both.
+                    if (rule.getSkipToTarget() != null && rule.getEndSurvey() != null) {
+                        rejectField(errors, "rule", "cannot have a skipTo target and an endSurvey property");
+                    } 
+                    // But must have either a skipTo target or an endSurvey property
+                    else if (rule.getSkipToTarget() == null && rule.getEndSurvey() == null) {
+                        rejectField(errors, "rule", "must have a skipTo target or an endSurvey property");
+                    }
+                    // Validate the endSurvey value cannot be false
+                    else if (Boolean.FALSE.equals(rule.getEndSurvey())) {
+                        rejectField(errors, "rule", "cannot set endSurvey to false");
+                    } 
+                    // Otherwise we can assume there's a skipToTarget, start checking that by looking for back references.
+                    else if (alreadySeenIdentifiers.contains(rule.getSkipToTarget())) {
                         rejectField(errors, "rule", "back references question %s", rule.getSkipToTarget());
```

Comment:
It doesn't look like there's currently a test for back references. How difficult would it be to add this?